File: assets/scripts/pages/menu_principal.py
"""Script de la page de menu principal."""


def on_page_enter(context):
    page = context.get("page", {})
    _log(f"Bienvenue sur {page.get('name')}")


def _log(message: str) -> None:
    print(f"[OptimaScript] {message}")

File: assets/scripts/pages/menu_principal.dart
import 'package:optimascript/api.dart';

Future<void> onPageEnter(ScriptContext context) async {
  await context.callHost(
    'log',
    positional: <Object?>['Ouverture du menu principal via l\'API Dart.'],
  );
}

Future<void> onPageLeave(ScriptContext context) async {
  await context.callHost(
    'log',
    positional: <Object?>['Fermeture du menu principal via l\'API Dart.'],
  );
}
